app/src/main/java/org/novinger/jason/demoreader/ListActivity.java

```diff
@@ -1,47 +1,70 @@
 package org.novinger.jason.demoreader;
 
+import android.os.StrictMode;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
 import android.view.View;
+import android.view.ViewPropertyAnimator;
 import android.widget.AdapterView;
 import android.widget.ArrayAdapter;
 import android.widget.ListView;
 
+import com.google.gson.Gson;
+
+import org.novinger.jason.demoreader.datamodels.Article;
+
+import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Collections;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 
 public class ListActivity extends AppCompatActivity {
 
+    private final static int qty = 50;
+    private final static String feedUrl = "http://www.dailydot.com/api/v1/content/article/";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_list);
 
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
+        Article[] articles;
+        try {
+            articles = getFeedItems(getFeedUrl());
+        } catch (IOException ex) {
+            articles = new Article[0];
+            Log.d("Exception", ex.toString());
+        }
+
         final ListView listview = (ListView) findViewById(R.id.listview);
-        String[] values = new String[] {
-                "Jason", "Cindy", "Hannah", "Leah", "Kirsten", "Matt", "Elle", "Trevor", "Alex",
-                "Eli", "Danika", "Lori", "Wyatt", "Chad"
-        };
-        final ArrayList<String> list = new ArrayList<String>();
-        Collections.addAll(list, values);
+        final ArrayList<String> list = new ArrayList<>();
+        for (Article article : articles) {
+            list.add(article.toString());
+        }
 
         final ArrayAdapter adapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, list);
         listview.setAdapter(adapter);
         listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, final View view, int position, long id) {
                 final String item = (String) parent.getItemAtPosition(position);
-                view.animate().setDuration(500).alpha(0)
-                    .withEndAction(new Runnable() {
-                        @Override
-                        public void run() {
-                            list.remove(item);
-                            adapter.notifyDataSetChanged();
-                            view.setAlpha(1);
-                        }
-                    });
+                final ViewPropertyAnimator viewPropertyAnimator = view.animate().setDuration(100).alpha(0);
+                viewPropertyAnimator.withEndAction(new Runnable() {
```

Comment:
Agree and I didn't like it at the time. Partly because I'm not a fan of the chain-able style anymore, but partly because I didn't completely understand what I was doing or why I needed it. Unsurprisingly, that bit came from a tutorial I was looking at as I did it.

This is removed in a subsequent PR.